README.md

```diff
@@ -25,34 +37,47 @@ Example installation:
 
 `spack install dla-future ^intel-mkl`
 
-Notice that, for the package to work correctly, the HPX option `max_cpu_count` must be set accordingly to the platform,
-as it represents the maximum number of OS-threads.
+Or you can go even further with a more detailed spec like this one, which builds dla-future in debug mode, using the clang compiler, specifying that the HPX on which it depends has to be built
+in debug mode too, with APEX instrumentation enabled, and that we want to use MPICH as MPI implementation, without fortran support (because clang does not support it).
+
+`spack install dla-future %clang build_type=Debug ^hpx build_type=Debug instrumentation=apex ^mpich ~fortran`
+
+Notice that, for the package to work correctly, the HPX option `max_cpu_count` must be set accordingly to the architecture, as it represents the size of the bitmask to interface with hardware threads.
 
 `spack install dla-future ^intel-mkl ^hpx max_cpu_count=256`
 
-## How to use the library
+#### Build the old good way
 
-Using DLAF in a CMake project is extremely easy!
+You can build all the dependencies by yourself, but you have to ensure that:
+- BLAS/LAPACK implementation is not multithreaded
+- HPX: `HPX_WITH_NETWORKING=none` + `HPX_WITH_MAX_CPU_COUNT=n` (according to number of cores in the architecture, suggested the next closest power of 2)
+- HPX and DLAF must have a compatible `CMAKE_BUILD_TYPE`: they must be built both in Debug, or with any combination of release types (Release, RelWithDebInfo or MinSizeRel)
+
+TODO configure cmake script for DLAF (with/without MKL)
```

Comment:
@rasolca @teonnik 
About this TODO, do you have any suggestion on how to deal with it? Do we want to explain all CMake variables of our project? Do we want to address it here or in another dedicated PR (in case I could open an issue about this)?

Reply:
I think these are at least worth listing. Most of the are self-explanatory. Perhaps something like this: 

- `DLAF_ASSERT_ENABLE` : set to ON|OFF to enable/disable cheap assertions (default: ON)
- `DLAF_ASSERT_MODERATE_ENABLE` : set to ON|OFF to enable/disable moderate assertions (default: ON in Debug, OFF otherwise)
- `DLAF_ASSERT_HEAVY_ENABLE` : set to ON|OFF to enable/disable heavy assertions (default: ON in Debug, OFF otherwise)

- `DLAF_WITH_CUDA` : enable CUDA support 
- `HPX_DIR`: ???
- `blaspp_DIR`: ???
- `lapackpp_DIR`: ???

`blaspp`/`lapackpp` dependency:
- `DLAF_WITH_MKL` (if blaspp/lapackpp is built with MKL)
- `MKL_ROOT`: ???

- `DLAF_BUILD_MINIAPPS` : set to ON|OFF to enable/disable building miniapps
- `DLAF_WITH_TEST` : set to ON|OFF to enable/disable building tests
- `DLAF_INSTALL_TESTS` : set to ON|OFF to enable/disable installing tests

Reply:
What about adding also `BUILD_DOC` (that might probably be renamed to `DLAF_BUILD_DOC` or similar)?

Reply:
I've just added a table trying to give an overview of all the options (it looks a bit cluttered to me, if you have any suggestion on how to improve it, it is absolutely welcome!).